### ISSUE TITLE
Disabling a Test Case in the profile does not prevent it from being run if called explicitly

### ISSUE THREAD

**Issue by tgreenx** [zonemaster/zonemaster-engine#1284]
The profile can be used to prevent Test Cases from being run (i.e. by removing them from the profile). This is true when Engine runs the whole test suite, or even specific Test modules, but not when running specific Test Cases (i.e., using [Zonemaster::Engine/test_method()](https://github.com/zonemaster/zonemaster-engine/blob/7af3518b067890247aad29c4de90c1d45ee9bdf2/lib/Zonemaster/Engine.pm#L73-L77) or [Zonemaster::Engine::Test/run_one()](https://github.com/zonemaster/zonemaster-engine/blob/7af3518b067890247aad29c4de90c1d45ee9bdf2/lib/Zonemaster/Engine/Test.pm#L155-L207)). That is because the checks are done directly in the `all()` method of each Test module (with the help of [Zonemaster::Engine::Util/should_run_test()](https://github.com/zonemaster/zonemaster-engine/blob/7af3518b067890247aad29c4de90c1d45ee9bdf2/lib/Zonemaster/Engine/Util.pm#L48-L53)). That method is not used when running specific Test Cases.

For example, using Zonemaster-CLI, after removing Address01 from the profile (note that Engine is started twice because of two `--test` arguments):
```
$ zonemaster-cli --show-testcase --test Address --test Address/address01 --level DEBUG --no-ipv6 --raw zonemaster.net | grep TEST_CASE_START
   0.00 DEBUG    ADDRESS02      TEST_CASE_START   testcase=address02
   3.33 DEBUG    ADDRESS03      TEST_CASE_START   testcase=address03
   0.00 DEBUG    ADDRESS01      TEST_CASE_START   testcase=address01
```

It could be intentional, but that seems odd to me. So, bug or feature? I would advocate for the former.

**Comment by matsduf** [comment]
Yes, Engine is started twice, but I noticed when preparing data for #1281 that cache and blacklisting survive from the first run to the second run. But I think it would be better if Engine could be run once.

I think it is fine that you can include a test case not in the profile if you explicitly calls it. What is the problem of that? What do you think should happen? And why?

**Comment by matsduf** [comment]
Resolved by #1294
